### PR TITLE
cmake: Only add `-Wall and -Werror` for our code.

### LINKED ISSUE #127: tool build failure: YAML::RegEx::m_z may be used uninitialized in this function
```
$ make
[  1%] Building CXX object third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o
In file included from /home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.cpp:1:0:
/home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.h: In constructor ‘YAML::RegEx::RegEx(const string&, YAML::REGEX_OP)’:
/home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.h:31:20: error: ‘<anonymous>.YAML::RegEx::m_z’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class YAML_CPP_API RegEx {
                    ^~~~~
cc1plus: all warnings being treated as errors
third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/build.make:518: recipe for target 'third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o' failed
make[2]: *** [third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o] Error 1
CMakeFiles/Makefile2:1873: recipe for target 'third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/all' failed
make[1]: *** [third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

```
$ gcc --version
gcc (Debian 7.3.0-5) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ cmake --version
cmake version 3.9.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

### PR DESCRIPTION
**Pull request by mithro**
Fixes #127.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,5 +27,9 @@ target_include_directories(yaml-cpp PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/third_party/yaml-cpp/include>
 	)
 
+# Set the CXX standard and compile time for our code only.
+set(CMAKE_CXX_STANDARD 14)
+add_compile_options(-Wall -Werror)
```
Comment (litghost) [inline CMakeLists.txt]: Prefer target_compile_options, which only modify specific targets.

Comment (litghost) [reply CMakeLists.txt]: See https://cmake.org/cmake/help/v3.0/command/target_compile_options.html

**Comment by mcmasterg** [comment]
Poke? Should I merge this or is someone going to refine it? Assuming @mithro is busy, propose @litghost refines it if he'd like the better target, otherwise I'd like to merge it in as its better than what we have now.

**Comment by mithro** [comment]
If @litghost has an alternative, merge that and close this. Otherwise, lets merge this and get @litghost to do a follow up?

**Pull request merged**